twi/twi.c

```diff
@@ -132,83 +69,38 @@ int twim_read_register(void * context, uint8_t address, uint8_t * buffer, uint32
+	while (! transfer_done);
 
 	transfer_done = false;
 	twim_err_code = nrfx_twim_rx(&m_twim, ICM_20948_ADD, buffer, length);
 	APP_ERROR_CHECK(twim_err_code);
 
-	wait_for_twi_event();
+	while (! transfer_done);
 
 	return (uint8_t)twim_err_code;
 }
 
 void int_pin_handler(nrfx_gpiote_pin_t pin, nrf_gpiote_polarity_t action)
 {
-	// needed the int pin handler, because with just PPI it was too fast WTF- no, ppi int cannot start twim for some fck reason
-	//		app_sched_event_put(NULL, 0, setup_fifo_burst);
-
+	// never happens after irqs were sorted out
 	if (app_sched_queue_space_get() > 20)
 	{
-		app_sched_event_put(NULL, 0, icm20948_service_isr);
+		twim_err_code = app_sched_event_put(NULL, 0, icm20948_service_isr);
+		APP_ERROR_CHECK(twim_err_code);
 	}
 	else
 	{
 		NRF_LOG_ERROR("dropped IMU sample");
 	}
 }
 
-//void setup_fifo_burst(void * p_event_data, uint16_t event_size)
-//{
-////	nrfx_gpiote_in_event_disable(INT1_PIN);
-//
-//	uint8_t buf[3];
-//	twim_read_register(NULL, 0x70, buf, 2); // fifo count
-//	twim_read_register(NULL, 0x1B, &buf[2], 1); // fifo overflow
-//	if (buf[2])
-//	{
-//		buf[0] = 0x1f; buf[1] = 0x1e;
-//		twim_write_register(NULL, 0x68, &buf[0], 1);
-//		twim_write_register(NULL, 0x68, &buf[1], 1);
-//	}
-//	NRF_LOG_INFO("fifo size: %d, %d", (uint16_t)(buf[0]<<8|buf[1]), buf[2]);
-////	twim_read_register(NULL, 0x72, twi_fifo_buffer, 255);
-//
-//	// setup non-blocking fifo read transfer and get task&event
-//	uint8_t fifo_buffer_address = 0x72;
-//	nrfx_twim_xfer_desc_t xfer = NRFX_TWIM_XFER_DESC_TXRX(ICM_20948_ADD, &fifo_buffer_address, 1, twi_fifo_buffer, TWI_FIFO_READ_LENGTH);
-//	uint32_t flags = 	NRFX_TWIM_FLAG_HOLD_XFER			|
-//						NRFX_TWIM_FLAG_RX_POSTINC       	|
-//						NRFX_TWIM_FLAG_NO_XFER_EVT_HANDLER	|
-//						NRFX_TWIM_FLAG_REPEATED_XFER;
-//	twim_err_code = nrfx_twim_xfer(&m_twim, &xfer, flags);
-//	if (!twim_err_code)
-//	{
-//		twi_task_addr = nrfx_twim_start_task_get(&m_twim, xfer.type);
-//		twi_stop_evt = nrfx_twim_stopped_event_get(&m_twim);
-//	}
-//
-//	twim_err_code = nrfx_ppi_channel_assign(ppi_channel_twim, twi_stop_evt, twi_task_addr);
-//	twim_err_code = nrfx_ppi_channel_fork_assign(ppi_channel_twim, timer_count_task);
-//
-////	twim_err_code = nrfx_ppi_channel_enable(ppi_channel_twim);
-//	twim_err_code = nrfx_ppi_group_enable(ppi_channel);
-//
-////	if (twim_err_code) return twim_err_code;
-////	twim_err_code = nrfx_ppi_channel_assign(ppi_channel_int, int_evt_addr, twi_task_addr);
-////	nrfx_ppi_channel_fork_assign(ppi_channel_int, out_task_addr);
-////	twim_err_code = nrfx_ppi_channel_enable(ppi_channel_int);
-//
-//	nrf_twim_task_trigger(m_twim.p_twim, twi_task_addr);
-//}
-
 ret_code_t twi_init (void)
 {
 	const nrfx_twim_config_t twim_config = {
 			.scl                = TWI_SCL,
 			.sda                = TWI_SDA,
 			.frequency          = NRF_TWIM_FREQ_400K, //0x0C380D40 - this will set it to 800kHz
-			.interrupt_priority = APP_IRQ_PRIORITY_LOW,
+			.interrupt_priority = 5,
```

Comment:
I think should be using a macro value (? 
It's going from a documented value to a magic one. Also not sure why this would be different from the default config value that is being provided in `sdk_config.h`

Reply:
I see the change was reverted instead of a define being used for setting the 5 value. `APP_IRQ_PRIORITY_LOW` will resolve to 6, my feedback was to use the define provided in `sdk_config.h:3659` instead and modify the default value for it to be 5